python/tk_hiero_export/sg_nuke_shot_export.py

```diff
@@ -261,7 +274,17 @@ def _beforeNukeScriptWrite(self, script):
             node.setName('ShotgunWriteNodePlaceholder')
 
             self.app.log_debug("Created ShotgunWriteNodePlaceholder Node: %s" % node._knobValues)
-            script.addNode(node)
+            # rather than using the script.addNode, we append our node directly to the nodeList
+            nodeList.append(node)
+
+            # now add our new node to the layout
+            currentLayoutContext.getNodes().append(node)
+
+        # now put back the viewer nodes layout
+        currentLayoutContext.getNodes().append(oldLayoutEnd)
```

Comment:
Solid work here, but one change I'd probably make would be to wrap the logic handling adding our placeholders in a try, and then reattach the stuff you disconnected in a finally block. That'll ensure it ALWAYS gets reattached, even if something goes wrong with our placeholder stuff.

That being said, if you don't think it's needed, then don't sweat it. If the situation is such that a failure of some kind to properly setup the placeholders means the export is hosed anyway, then a try/finally might not be necessary.

Reply:
Good idea, I've put that in and logged an exception, thanks!